src/main/java/org/geysermc/discordbot/commands/moderation/ForumCommand.java

```diff
@@ -0,0 +1,304 @@
+                    matchFound = true;
+                    break;
+                }
+            }
+
+            if (matchFound) {
+                List<ForumTag> currentTags = event.getChannel().asThreadChannel().getAppliedTags();
+                ForumTagSnowflake newTag = ForumTagSnowflake.fromId(tags.get(index).getId());
+                List<ForumTagSnowflake> updatedTags = new ArrayList<>(currentTags);
+                updatedTags.add(newTag);
+                ThreadChannelManager manager = event.getChannel().asThreadChannel().getManager().setAppliedTags(updatedTags);
+                manager.queue(
+                        unused -> event.reply("Post is tagged with " + tag + ".").queue(),
+                        error -> event.reply("Could not tag post").queue()
+                );
+
+            } else {
+                event.reply("No matching tag found for " + tag + ".").queue();
+            }
+        }
+    }
+
+    public static class removeTagPostSubCommand extends SlashCommand {
+        public removeTagPostSubCommand() {
+            this.name = "remove-tag";
+            this.help = "Remove a tag from post";
+            this.userPermissions = new Permission[]{Permission.CREATE_PUBLIC_THREADS};
+            this.options = getTags();
+        }
+
+        @Override
+        protected void execute(@NotNull SlashCommandEvent event) {
+            if (!(event.getChannel() instanceof ThreadChannel) || !event.getChannel().asThreadChannel().getParentChannel().getId().equals(FORUM_CHANNEL_ID)) {
+                event.reply("Command can only be used in forum channels.").queue();
+                return;
+            }
+
+            String tagToRemove = event.optString("tag");
+            int indexToRemove = -1;
+            List<ForumTag> currentTags = event.getChannel().asThreadChannel().getAppliedTags();
+            for (int i = 0; i < currentTags.size(); i++) {
+                ForumTag currentTag = currentTags.get(i);
+                if (currentTag.getName().equalsIgnoreCase(tagToRemove)) {
+                    indexToRemove = i;
+                    break;
+                }
+            }
+
+            if (indexToRemove != -1) {
+                List<ForumTagSnowflake> updatedTags = new ArrayList<>(currentTags);
+                updatedTags.remove(indexToRemove);
+                ThreadChannelManager manager = event.getChannel().asThreadChannel().getManager().setAppliedTags(updatedTags);
+                manager.queue(
+                        unused -> event.reply("Tag " + tagToRemove + " removed.").queue(),
+                        error -> event.reply("Could not remove tag from post").queue()
+                );
+
+            } else {
+                event.reply("Post was not tagged with  " + tagToRemove + ".").queue();
+            }
+        }
+    }
+
+    @NotNull
+    public static List<OptionData> getTags() {
+        return List.of(
+                new OptionData(OptionType.STRING, "tag", "name of the tag")
```

Comment:
Is it possible to retrieve the parent forum channel's tags?

Reply:
> Is it possible to retrieve the parent forum channel's tags?

Not at this point sadly